### ISSUE TITLE
Assertions check

### ISSUE THREAD

**Issue by DominicDetta** [zendframework/zend-permissions-rbac#32]
I don't understand why, in the case of an assertion defined, there is no check whether the role has the permission.

Assume we have the following code: 

```php
<?php
 use Zend\Permissions\Rbac\AssertionInterface;
 use Zend\Permissions\Rbac\Rbac;

 class AssertExample implements AssertionInterface
 {
     public function assert(Rbac $rbac)
     {
         return true;
     }
 }

// Adding new role but no permissions
 $rbac = new Rbac();
 $rbac->addRole('admin');

 $assertion = new AssertExample();

 //  Permission not granted
 if ($rbac->isGranted('admin', 'edit.settings')) {
 }

 // true always - even if the role is not associated with the permission!
 if ($rbac->isGranted('admin', 'edit.settings', $assertion)) {
 }
```
Can you please be so kind to explain to me?

**Comment by ezimuel** [comment]
@DominicDetta the dynamic assertion using `AssertionInterface` has been fixed in [develop branch](https://github.com/zendframework/zend-permissions-rbac/tree/develop) [here](https://github.com/zendframework/zend-permissions-rbac/blob/develop/src/Rbac.php#L142-L158). Basically, we return the `AssertionInterface::assert` AND `getRole()->hasPermission()`. 
With this change,  the last check of your case will be false, because `admin` does not have `edit.settings` permission.
We need to release a new version of zend-premissions-rbac, I'll do asap. Thanks for you comment!

**Comment by DominicDetta** [comment]
@ezimuel Thanks for the explanation. It was only my mistake to not verify the develop branch. Now all makes sense.

**Comment by ezimuel** [comment]
@DominicDetta we just released zend-permissions-rbac [2.6.0](https://github.com/zendframework/zend-permissions-rbac/releases/tag/release-2.6.0) containing the fix for dynamic assertion.

**Comment by DominicDetta** [comment]
Awesome! I saw yesterday when I did a `composer update`. Thank you!!!
